File: C/seleccion_binaria.c
// Programa para ordenar mediante selection sort y buscar mediante busqueda binaria

#include <stdio.h>

// Funcion para llenar un arreglo

void fill_array(int arr[], int size)
{
    register int i;
    int element;

    for (i = 0; i < size; i++)
    {
        printf("\nIngrese el valor del elemento %d ", i);
        scanf("%d", &element);
        arr[i] = element;
    }
    printf("\n");
}

// Funcion para imprimir un arreglo

void print_array(int arr[], int size)
{
    register int i;

    printf("\nLos elementos del arreglo son: [ ");
    
    for (i = 0; i < size; i++)
    {
        printf("%d ",arr[i]);
    }
    
    printf("]\n");
}

// Funcion del selection sort

void selection_sort(int arr[], int size)
{
    register int i, j;
    int min;

    for (i = 0; i < size - 1; i++)
    {
        min = i;
        for (j = i + 1; j < size; j++)
        {
            if (arr[j] < arr[min])
                min = j;
        }
        
        if (min != i)
        {
            int aux;

            aux = arr[i];
            arr[i] = arr[min];
            arr[min] = aux;
        }
    }
}

// Funcion para busqueda binaria

int binary_search(int arr[], int size, int v)
{
    int start, end, mid;
    start = 0;
    end = size - 1;

    while (start <= end)
    {
        mid = (start + end) / 2;

        if (arr[mid] == v)
            return mid;
        else if (arr[mid] > v)
            end = mid - 1;
        else
            start = mid + 1;
    }

    return -1;
}

void main()
{
    int n, pos, value;

    printf("\nIngrese la cantidad de elementos del arreglo: ");
    scanf("%d", &n);

    int array[n];

    // Llamada a funcion para rellenar el arreglo

    fill_array(array, n);
    print_array(array, n);

    // Llamada a funcion para ordenar mediante selection sort

    printf("\nOrdenando mediante selection sort...\n");
    selection_sort(array, n);
    print_array(array, n);

    // Llamada a funcion para la busqueda binaria

    printf("\nIngrese el valor del elemento a buscar: ");
    scanf("%d", &value);

    pos = binary_search(array, n, value);
    printf("\nLa posicion del elemento es: %d\n", pos);

}

File: Python/seleccion_binaria.py
# Programa para ordenar una lista mediante selection sort y luego buscar un elemento mediante binary search

# Funcion para llenar una lista

def fill_list(lista):
    length = int(input("\nIngrese la cantidad de elementos de la lista: "))

    for i in range(0, length):
        value = int(input(f"Ingrese el elemento numero {i}: "))
        lista.append(value)

# Funcion para el selection sort

def selection_sort(lista, size):

    for i in range(0, size - 1):
        min = i
        for j in range(i + 1, size):
            if lista[j] < lista[min]:
                min = j

        if min != i:
            aux = lista[i]
            lista[i] = lista[min]
            lista[min] = aux

# Funcion para la binary search

def binary_search(lista, size, v):
    start = 0
    end = size - 1

    while start <= end:
        mid = (start + end) // 2

        if lista[mid] == v:
            return mid
        elif lista[mid] < v:
            start = mid + 1
        else:
            end = mid - 1
    
    return -1


# Funcion main()

int_list = []

# Llamada a funcion para llenar la lista

fill_list(int_list)
list_size = len(int_list)
print(f"\nLa lista es: {int_list}")

# Llamada a funcion para ordenar la lista mediante selection sort

print("\nOrdenando con selection sort...\n")
selection_sort(int_list, list_size)
print(f"La lista es: {int_list}\n")

# Llamada a funcion para la binary search

value = int(input("Ingrese el valor del elemento a buscar: "))
pos = binary_search(int_list, list_size, value)
print(f"\nLa posicion del elemento {value} es: {pos}\n")